tests/tamr_client/fake.py

```diff
@@ -27,7 +47,28 @@ def _to_kwargs(fake):
         path = req.get("path")
         url = "http://localhost/api/versioned/v1/" + path
 
-    return dict(method=req["method"], url=url, **resp)
+    # Get response body from either ndjson or json
+    if resp.get("ndjson") is not None:
+        resp["body"] = "\n".join((dumps(line) for line in resp["ndjson"]))
+    elif resp.get("json") is not None:
+        resp["body"] = dumps(resp["json"])
+
+    # Get expected request body from ndjson
+    if req.get("ndjson") is not None:
+        req["body"] = [x for x in req["ndjson"]]
+    elif req.get("json") is not None:
+        req["body"] = req["json"]
+
+    rsps.add_callback(
+        method=req["method"],
+        url=url,
+        callback=partial(
```

Comment:
One organizational alternative is to have `check_request_body` just take the `request` and `expected_body` as args and raise exception if they don't match. Then the rest of the logic (passing through `status` and `response_json`) can be encapsulated in another function, say `_callback`:

```python
def _callback(request, expected_body, status, response_json):
    check_request_body(request, expected_body)
    return status, {}, response_json
```

I bring this up only because the name `check_request_body` implies that this function should *only* perform a check and have no side-effects or need to take the `status` or `response_json` as args.

Another nit: we should probably put a leading prefix on `_check_response_body` in any case.